dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/ModuleSelectionConfigImpl.java

```diff
@@ -218,6 +327,17 @@ public String toString() {
         }
     }
 
+    @Override
+    public synchronized void processConfigUpdate() {
+        //Reset everything for the next lazyInit
+        hasBeenInit = false;
+        isAll = false;
+        isFirst = false;
+        serverxmlmode = false;
+        included = null;
+        excluded = null;
+    }
```

Comment:
We have an issue with thread safety because we have this method to reset everything and re-initialize on next use.

If `lazyInit` was only called once it would be sufficient for only the `lazyInit` method to be synchronized as long as every method which needs to access any of these values calls `lazyInit` before doing so.

However, since we have the ability to reset, every use of any of these variables must be synchronized, otherwise they could be reset in the middle of a method which uses them.

To avoid this, I would suggest:
- add a new static inner class to hold these values - let's call it `ConfigValues`
- rather than having a `lazyInit` method which sets all these fields, have a method which creates, caches and returns a `ConfigValues` object
  - all other code which needs the `ConfigValues` object must retrieve it via this method
- `processConfigUpdate` then only needs to delete the cached `ConfigValues` object
- this method and the method to get `ConfigValues` may need synchronization or correct use of `volatile`, but everything else should be correct without synchronization

Reply:
All sounds good, though I think the inner field should not be static since every ConfigValue object will be associated with an outer ModuleSelectionConfigImpl object that it will occasionally have to reference to access the OpenAPIAppConfigProvider.

Reply:
I was imagining that the outer class would extract the config and pass all those values into the inner class so it wouldn't need to  access the `OpenAPIAppConfigProvider`, but making the inner class non-static so it can access that class itself also seems fine.

However, an important point is that the inner object is immutable so that its values won't change once a method has retrieved it.

If the config gets updated, we would create a new instance of the inner object and future calls would get the new object.